crates/neon-runtime/src/napi/string.rs

```diff
@@ -1,7 +1,47 @@
-use raw::{Local, Isolate};
+use std::mem::MaybeUninit;
+use std::ptr;
 
-pub unsafe extern "C" fn new(_out: &mut Local, _isolate: *mut Isolate, _data: *const u8, _len: i32) -> bool { unimplemented!() }
+use nodejs_sys as napi;
 
-pub unsafe extern "C" fn utf8_len(_str: Local) -> isize { unimplemented!() }
+use raw::{Env, Local};
 
-pub unsafe extern "C" fn data(_out: *mut u8, _len: isize, _str: Local) -> isize { unimplemented!() }
+pub unsafe fn new(out: &mut Local, env: Env, data: *const u8, len: i32) -> bool {
+    let status = napi::napi_create_string_utf8(
+        env,
+        data as *const i8,
+        len as usize,
+        out,
+    );
+
+    status == napi::napi_status::napi_ok
+}
+
+pub unsafe extern "C" fn utf8_len(env: Env, value: Local) -> isize {
+    let mut len = MaybeUninit::uninit();
+    let status = napi::napi_get_value_string_utf8(
+        env,
+        value,
+        ptr::null_mut(),
+        0,
+        len.as_mut_ptr(),
+    );
+
+    assert_eq!(status, napi::napi_status::napi_ok);
+
+    len.assume_init() as isize
+}
+
+pub unsafe extern "C" fn data(env: Env, out: *mut u8, len: isize, value: Local) -> isize {
+    let mut read = MaybeUninit::uninit();
+    let status = napi::napi_get_value_string_utf8(
+        env,
+        value,
+        out as *mut i8,
+        len as usize,
```

Comment:
There's a ton of casting going on in these string methods. Should we eliminate the casting in the `runtime` methods and branch further in the higher level `neon` methods?

My intuition is that we should keep them the same and flag them for follow-up since `runtime` is an internal crate.

Reply:
https://github.com/neon-bindings/neon/issues/472